subiquity/server/controllers/snaplist.py

```diff
@@ -50,22 +54,51 @@ def __init__(self, model, snapd, store_section, context):
         self.context = context
 
         self.main_task = None
-        self.snap_list_fetched = False
-        self.failed = False
 
         self.snapd = snapd
         self.pending_snaps = []
         self.tasks = {}  # {snap:task}
 
+        self.load_list_task_created = asyncio.Event()
+
+    def fetch_list_completed(self) -> bool:
```

Comment:
Refactored, thanks.
I also removed the `if task is None` block since it was only useful when a `SingleInstanceTask` was used.
`SingleInstanceTask` have their `.task` set to `None` until `.start_sync()` is called.